app/controllers/application_controller.rb

```diff
@@ -2,4 +2,17 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :null_session, if: Proc.new { |c| c.request.format == 'application/json' }
+
+  def authenticate_user_from_token!
+    auth_token = request.headers['X-AUTH-TOKEN']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.